node/node.go

```diff
@@ -374,76 +337,71 @@ func (node *Node) Start() error {
 			return err
 		}
 		topicNamed := allTopics[i].Name
+
+		if nodeconfig.GroupID(topicNamed) == nodeconfig.NewGroupIDByShardID(
+			nodeconfig.ShardID(node.Consensus.ShardID),
+		) {
+
+			pubsub.RegisterTopicValidator(
+				topicNamed,
+				func(ctx context.Context, peer libp2p_peer.ID, msg *libp2p_pubsub.Message) bool {
+					sig2 := msg.GetData()
+					if len(sig2) < p2pMsgPrefixSize {
+						return true
+					}
+
+					cnsMsg := sig2[p2pMsgPrefixSize:][proto.MessageCategoryBytes:]
+
+					var (
+						m         msg_pb.Message
+						senderKey *bls.PublicKey
+					)
+
+					if err := protobuf.Unmarshal(cnsMsg, &m); err != nil {
+						errChan <- err
+						return false
+					}
+
+					if c := m.GetConsensus(); c.GetSenderPubkey() != nil {
+						key, err := bls_cosi.BytesToBLSPublicKey(
+							c.GetSenderPubkey(),
+						)
+						if err != nil {
+							errChan <- err
+							return false
+						}
+						senderKey = key
+					}
+
+					if err := consensus.VerifyMessageSig(senderKey, &m); err != nil {
+						pubsub.BlacklistPeer(peer)
+						errChan <- err
+						return false
+					}
+
+					return true
+
+				},
+				libp2p_pubsub.WithValidatorTimeout(24),
```

Comment:
maybe we can define some configuration for those critical parameters.

Reply:
Not necessarily, they are part of a bigger config already proper via ipfs. I will unify then, when will bring that other pr back up to speed